Framework/Core/src/PropertyTreeHelpers.cxx

```diff
@@ -98,10 +99,11 @@ std::vector<T> toVector(std::string const& input)
   std::vector<T> result;
   //check if the array string has correct array type symbol
   assert(input[0] == variant_array_symbol<T>::symbol);
-  //strip type symbol and parentheses
-  boost::tokenizer<> tokenizer(input.substr(2, input.size() - 3));
-  for (auto it = tokenizer.begin(); it != tokenizer.end(); ++it) {
-    result.push_back(boost::lexical_cast<T>(*it));
+  std::regex nmatch(R"((?:(?!=,)|(?!=\[))[+-]?\d+\.?\d*(?:[eE][+-]?\d+)?(?=,|\]))", std::regex_constants::ECMAScript);
```

Comment:
Do you still need the tokenizer? iF not, can you remove the headers?

Reply:
Done.